hack/actions/gen/main.go

```diff
@@ -0,0 +1,743 @@
+//
+// Copyright (C) 2019-2024 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"text/template"
+	"time"
+
+	"github.com/vdaas/vald/internal/file"
+	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/safety"
+	"github.com/vdaas/vald/internal/sync/errgroup"
+	"gopkg.in/yaml.v2"
+)
+
+var license string = `#
+# Copyright (C) 2019-{{.Year}} {{.Maintainer}}
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# DO_NOT_EDIT this workflow file is generated by https://github.com/vdaas/vald/blob/main/hack/actions/gen/main.go
+`
+
+var licenseTmpl *template.Template = template.Must(template.New("license").Parse(license))
+
+type (
+	Workflow struct {
+		Name string `yaml:"name"`
+		On   On     `yaml:"on"`
+		Jobs Jobs   `yaml:"jobs"`
+	}
+
+	On struct {
+		Schedule          Schedule          `yaml:"schedule,omitempty"`
+		Push              Push              `yaml:"push"`
+		PullRequest       PullRequest       `yaml:"pull_request"`
+		PullRequestTarget PullRequestTarget `yaml:"pull_request_target"`
```

Comment:
It might be better to unify the Paths structure!
The following is an example.

```
	On struct {
		Schedule          Schedule          `yaml:"schedule,omitempty"`
		Push              Push                    `yaml:"push"`
		PullRequest       Paths              `yaml:"pull_request"`
		PullRequestTarget Paths          `yaml:"pull_request_target"`

	Paths []string
```

Reply:
applied it!